OpenQA/Isotovideo/Utils.pm

```diff
@@ -53,6 +55,33 @@ sub git_remote_url ($git_repo_dir) {
     return 'UNKNOWN (error on git remote call)';    # uncoverable statement
 }
 
+sub _handle_caching ($clone_url, $clone_depth, $branch, $clone_cmd, $handle_output) {
+    # determine cache directory and ensure its parent directory exists
+    return undef unless my $git_cache_dir = $bmwqemu::vars{GIT_CACHE_DIR};
+    my $cache_dir = path($git_cache_dir, $clone_url->path);
+    path($git_cache_dir, $clone_url->path->to_dir)->make_path;
+
+    # lock the cache directory
+    open(my $lock, '>', "$cache_dir.lock") or die "Unable to open lock file '$cache_dir.lock' for Git caching: $!\n";
+    flock($lock, LOCK_EX) or die "Unable to lock '$cache_dir.lock' for Git caching: $!\n";
+    my $lock_guard = scope_guard sub {
+        flock($lock, LOCK_UN) or die "Unable to unlock '$cache_dir.lock' after Git caching: $!\n";
+        close($lock);
+    };
+
+    # clone new bare repo if none exists so far
+    unless (-e $cache_dir) {
+        bmwqemu::fctinfo "Creating bare repository for caching '$clone_url' under '$cache_dir'";
+        $handle_output->($?, qx{$clone_cmd --bare --depth='$clone_depth' '$clone_url' '$cache_dir' 2>&1});
+    }
+
+    # fetch required branch specifically or just fetch new refs if no branch specified
+    bmwqemu::fctinfo "Updating Git cache for '$clone_url' under '$cache_dir'";
+    $branch = $branch ? "$branch" : '';
```

Comment:
What is that line supposed to do? Why the quotes `"$branch"`?
Or maybe you wanted:
```suggestion
    $branch = $branch ? qq{"$branch"} : '';
```